extensions/wikia/AchievementsII/services/AchAwardingService.class.php

```diff
@@ -44,11 +44,16 @@ public function migration($user_id)  {
 	public function awardCustomNotInTrackBadge($user, $badge_type_id) {
 		wfProfileIn(__METHOD__);
 
-		global $wgExternalSharedDB;
+		global $wgExternalSharedDB, $wgAchievementsEditAddPhotoOnly;
+
+		if ( !empty( $wgAchievementsEditAddPhotoOnly ) && ( $badge_type_id != BADGE_EDIT && $badge_type_id != BADGE_PICTURE ) ) {
+			wfProfileOut(__METHOD__);
+			return;
+		}
 
 		$this->mUser = $user;
 
-		if( self::canEarnBadges( $this->mUser ) ) {
+		if ( self::canEarnBadges( $this->mUser ) ) {
```

Comment:
What's going on with those extra spaces? It seems like you are trying to format code properly but it's not consistent:
```if ( self::canEarnBadges( $this->mUser ) ) {```
```if ($badge === false) {```
```if(!empty($wgAchievementsEditAddPhotoOnly)) {```